src/gui/DatabaseOpenWidget.cpp

```diff
@@ -212,7 +215,34 @@ void DatabaseOpenWidget::openDatabase()
     QApplication::restoreOverrideCursor();
     m_ui->passwordFormFrame->setEnabled(true);
 
-    if (!ok) {
+    if (ok) {
+#ifdef WITH_XC_TOUCHID
+        QHash<QString, QVariant> useTouchID = config()->get("UseTouchID").toHash();
+
+        // check if TouchID can & should be used to unlock the database next time
+        if (m_ui->checkTouchID->isChecked() && TouchID::getInstance().isAvailable()) {
+            // encrypt and store key blob
+            if (TouchID::getInstance().storeKey(m_filename, PasswordKey(m_ui->editPassword->text()).rawKey())) {
+                useTouchID.insert(m_filename, true);
+            }
+        } else {
+            // when TouchID not available or unchecked, reset for the current database
+            TouchID::getInstance().reset(m_filename);
+            useTouchID.insert(m_filename, false);
+        }
+
+        config()->set("UseTouchID", useTouchID);
+#endif
+
+        if (m_ui->messageWidget->isVisible()) {
+            m_ui->messageWidget->animatedHide();
+        }
+
+        emit dialogFinished(true);
+        m_isOpeningDatabase = false;
+        clearForms();
+    } else {
+        m_isOpeningDatabase = false;
```

Comment:
Shouldn't this be a Mutex then as well?

Reply:
Perhaps, I'm second guessing the use of mutex